src/gaxios.ts

```diff
@@ -40,6 +40,19 @@ function hasFetch() {
   return hasWindow() && !!window.fetch;
 }
 
+function hasBuffer() {
+  return typeof Buffer !== 'undefined';
+}
+
+function hasHeader(options: GaxiosOptions, headerCheck: string) {
```

Comment:
Support both `content-type` and `Content-Type`, and `content-Type`, when deciding whether to reset header.